lisa/util/shell.py

```diff
@@ -290,25 +299,55 @@ def spawn(
     ) -> spur.ssh.SshProcess:
         self.initialize()
         assert self._inner_shell
+        have_tried_minimal_type = False
 
-        try:
-            process: spur.ssh.SshProcess = _spawn_ssh_process(
-                self._inner_shell,
-                command=command,
-                update_env=update_env,
-                store_pid=store_pid,
-                cwd=cwd,
-                stdout=stdout,
-                stderr=stderr,
-                encoding=encoding,
-                use_pty=use_pty,
-                allow_error=allow_error,
-            )
-        except FunctionTimedOut:
-            raise LisaException(
-                f"The remote node is timeout on execute {command}. "
-                f"It may be caused by paramiko/spur not support the shell of node."
-            )
+        while True:
+            try:
+                if self._inner_shell._spur._shell_type == spur.ssh.ShellTypes.minimal:
+                    # minimal shell type doesn't support store_pid
+                    store_pid = False
```

Comment:
Because the pid is not saved, so the process cannot be killed. Any side effect here?

Reply:
From my understanding, only when the process timeout, process.kill() is called, then pid is needed. 
So far, in my local testing, I haven't seen any timeout yet. If there is timeout, the worst result might be one process in Linux couldn't be killed, it should not have an impact on the test.